Tests/TuistSupportTests/Extensions/URL+ExtraTests.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+import XCTest
+import TuistSupport
+
+// I (danibachar) have added these tests to show the fact that this validation might not be the best approach
+// One might want to migrate to Regex validation
+
+final class URLExtraConsumerTests: XCTestCase {
+    func test_valid_url() {
+        XCTAssertTrue(URL.isValid("https://google.com"))
+        XCTAssertTrue(URL.isValid("https:/google.com"))
+        XCTAssertTrue(URL.isValid("https:/google"))
+        XCTAssertTrue(URL.isValid("https:google.com"))
+        XCTAssertTrue(URL.isValid("ws://google.com"))
+        XCTAssertTrue(URL.isValid("wss://google.com:443"))
+        XCTAssertTrue(URL.isValid("google.com"))
+        XCTAssertTrue(URL.isValid("google"))
+        XCTAssertTrue(URL.isValid("1"))
+    }
+    
+    func test_non_valid_url_empty() {
+        XCTAssertFalse(URL.isValid(""))
+    }
+}
```

Comment:
This is not supposed to be merged, right?

Reply:
I didn't see any tests for this API and thought of adding those, I can definitely remove them / migrate to another validation approach

Reply:
I would rather just remove this validation entirely as it looks quite useless :D

Reply:
Or use a regex as you said, but probably not worth the effort

Reply:
I tend to agree .
If its ok with you, I'm going with your first suggestion of removing it altogether. This validation is only used as part of the linter under `Sources/TuistGenerator/Linter/PackageLinter.swift`.

My line of thought here is that SPM supports different types of URLs that i'm not even sure Ill be able to cover (https/http/git@ for ssh etc)

I think that delegating the verification to SPM is the best approach